hazelcast-jet-core/src/main/java/com/hazelcast/jet/core/Processor.java

```diff
@@ -234,13 +233,7 @@ default boolean finishSnapshotRestore() {
      * Context passed to the processor in the
      * {@link #init(Outbox, Context) init()} call.
      */
-    interface Context {
-
-        /**
-         * Returns the current Jet instance
-         */
-        @Nonnull
-        JetInstance jetInstance();
+    interface Context extends ProcessorSupplier.Context {
 
         /**
          *  Return a logger for the processor
```

Comment:
The `logger()` is also in `PS.Context`, we can delete it here maybe. Is it only due to javadoc?
Also, the `processingGuarantee` and `snapshottingEnabled` methods can be pushed up to PMS level.

Reply:
Yes they end up all being the same thing. Only ProcCtx has the unique index so maybe we should just have two types rather than three?